tests/spin/test_lattice.py

```diff
@@ -0,0 +1,541 @@
+    assert len(lattice.lattice_points == expected_number)
+
+
+@pytest.mark.parametrize(
+    # expected_edges here were obtained manually.
+    ("vectors", "positions", "n_cells", "boundary_condition", "expected_edges"),
+    [
+        (
+            [[0, 1], [1, 0]],
+            [[0, 0]],
+            [3, 3],
+            [True, True],
+            [
+                (0, 1, 0),
+                (1, 2, 0),
+                (3, 4, 0),
+                (5, 8, 0),
+                (6, 8, 0),
+                (0, 3, 0),
+                (1, 4, 0),
+                (0, 6, 0),
+                (4, 7, 0),
+                (6, 7, 0),
+                (1, 7, 0),
+                (0, 2, 0),
+                (4, 5, 0),
+                (3, 6, 0),
+                (2, 5, 0),
+                (3, 5, 0),
+                (7, 8, 0),
+                (2, 8, 0),
+            ],
+        ),
+        (
+            [[0, 1], [1, 0]],
+            [[0, 0]],
+            [3, 4],
+            [True, False],
+            [
+                (3, 7, 0),
+                (8, 9, 0),
+                (0, 8, 0),
+                (1, 9, 0),
+                (4, 5, 0),
+                (5, 6, 0),
+                (4, 8, 0),
+                (5, 9, 0),
+                (9, 10, 0),
+                (0, 1, 0),
+                (10, 11, 0),
+                (0, 4, 0),
+                (1, 2, 0),
+                (1, 5, 0),
+                (2, 10, 0),
+                (6, 7, 0),
+                (6, 10, 0),
+                (3, 11, 0),
+                (2, 3, 0),
+                (2, 6, 0),
+                (7, 11, 0),
+            ],
+        ),
+        (
+            [[1, 0], [0.5, np.sqrt(3) / 2]],
+            [[0.5, 0.5 / 3**0.5], [1, 1 / 3**0.5]],
+            [2, 2],
+            True,
+            [
+                (0, 1, 0),
+                (1, 2, 0),
+                (2, 7, 0),
+                (0, 3, 0),
+                (1, 4, 0),
+                (2, 3, 0),
+                (6, 7, 0),
+                (4, 5, 0),
+                (5, 6, 0),
+                (0, 5, 0),
+                (3, 6, 0),
+                (4, 7, 0),
+            ],
+        ),
+        (
+            [[1, 0], [0.5, np.sqrt(3) / 2]],
+            [[0.5, 0.5 / 3**0.5], [1, 1 / 3**0.5]],
+            [2, 2],
+            False,
+            [
+                (0, 1, 0),
+                (1, 2, 0),
+                (1, 4, 0),
+                (2, 3, 0),
+                (3, 6, 0),
+                (4, 5, 0),
+                (5, 6, 0),
+                (6, 7, 0),
+            ],
+        ),
+    ],
+)
+def test_boundary_condition(vectors, positions, n_cells, boundary_condition, expected_edges):
+    r"""Test that the correct edges are obtained for given boundary conditions"""
+    lattice = Lattice(
+        n_cells=n_cells, vectors=vectors, positions=positions, boundary_condition=boundary_condition
+    )
+    assert sorted(lattice.edges) == sorted(expected_edges)
```

Comment:
There is randomness in the order that forces us to sort?

Reply:
Yes, the edges calculated by KDTree use depth first search algorithm whereas when I manually got them, I just created the pictures and added edge by edge.